### PR TITLE
Patch to enhance multicast script

### PR DESCRIPTION
**Pull request by hathyaga**
This patch will add a multicast route to the peer to
ensure that the multicasts to the peer will go via
this route from the test interface

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
job log: 

linux-hxu5:~/automation-avocado/avocado-misc-tests/io/net # avocado run multicast.py -m multicast.py.data/multicast.yaml --show-job-log 
found extension EntryPoint.parse('journal = avocado.plugins.journal:JournalResult')
found extension EntryPoint.parse('tap = avocado.plugins.tap:TAPResult')
found extension EntryPoint.parse('human = avocado.plugins.human:Human')
found extension EntryPoint.parse('teststmpdir = avocado.plugins.teststmpdir:TestsTmpDir')
found extension EntryPoint.parse('jobscripts = avocado.plugins.jobscripts:JobScripts')
found extension EntryPoint.parse('human = avocado.plugins.human:HumanJob')
found extension EntryPoint.parse('yaml_to_mux = avocado_varianter_yaml_to_mux:YamlToMux')
Commands configured by file: /etc/avocado/sysinfo/commands
Files configured by file: /etc/avocado/sysinfo/files
Profilers configured by file: /etc/avocado/sysinfo/profilers
Profiler disabled
Command line: /usr/bin/avocado run multicast.py -m multicast.py.data/multicast.yaml --show-job-log

Avocado version: 56.0

Config files read (in order):
/etc/avocado/avocado.conf
/etc/avocado/conf.d/gdb.conf
/root/.config/avocado/avocado.conf

Avocado config:
Section.Key                             Value
datadir.paths.base_dir                  /var/lib/avocado
datadir.paths.test_dir                  /usr/share/avocado/tests
datadir.paths.data_dir                  /var/lib/avocado/data
datadir.paths.logs_dir                  ~/avocado/job-results
sysinfo.collect.enabled                 True
sysinfo.collect.commands_timeout        -1
sysinfo.collect.installed_packages      False
sysinfo.collect.profiler                False
sysinfo.collect.locale                  C
sysinfo.collect.per_test                False
sysinfo.collectibles.commands           /etc/avocado/sysinfo/commands
sysinfo.collectibles.files              /etc/avocado/sysinfo/files
sysinfo.collectibles.profilers          /etc/avocado/sysinfo/profilers
runner.output.colored                   True
runner.output.utf8                      
remoter.behavior.reject_unknown_hosts   False
remoter.behavior.disable_known_hosts    False
job.output.loglevel                     debug
restclient.connection.hostname          localhost
restclient.connection.port              9405
restclient.connection.username          
restclient.connection.password          
plugins.disable                         []
plugins.skip_broken_plugin_notification []
plugins.loaders                         ['file', '@DEFAULT']
gdb.paths.gdb                           /usr/bin/gdb
gdb.paths.gdbserver                     /usr/bin/gdbserver

Avocado Data Directories:

base     /var/lib/avocado
tests    /usr/share/avocado/tests
data     /var/lib/avocado/data
logs     /root/avocado/job-results/job-2017-11-28T01.45-2b436b5

Multiplex tree representation:
 \-- run
      \-- Parameters

Multiplex variants (1):
Variant Parameters-21f4:    /run/Parameters
Temporary dir: /var/tmp/avocado_4yn9Oe

Job ID: 2b436b5477ddcc1d06e6b8c8492304551d432b67

Not logging /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor (file does not exist)
Not logging /proc/pci (file does not exist)
Commands configured by file: /etc/avocado/sysinfo/commands
Files configured by file: /etc/avocado/sysinfo/files
Profilers configured by file: /etc/avocado/sysinfo/profilers
Profiler disabled
PARAMS (key=timeout, path=*, default=None) => None
START 1-multicast.py:ReceiveMulticastTest.test_multicast;Parameters-21f4
Test metadata:
  filename: /root/automation-avocado/avocado-misc-tests/io/net/multicast.py
Zypper version: 1.13.37
Running 'rpm -q net-tools'
[stdout] net-tools-2.0+git20170221.479bb4a-1.287.ppc64le
Command 'rpm -q net-tools' finished with 0 after 0.0107679367065s
Running 'rpm -q openssh'
[stdout] openssh-7.2p2-2.5.ppc64le
Command 'rpm -q openssh' finished with 0 after 0.0116300582886s
Running 'rpm -q iputils'
[stdout] iputils-s20161105-1.6.ppc64le
Command 'rpm -q iputils' finished with 0 after 0.0106949806213s
PARAMS (key=interface, path=*, default=None) => 'eth1'
PARAMS (key=peer_ip, path=*, default=) => '90.90.90.201'
PARAMS (key=user_name, path=*, default=root) => 'root'
Running 'ssh root@90.90.90.201 "ip addr show  | grep 90.90.90.201 | grep -oE '[^ ]+$'"'
[stdout] enp128s0f4d2
Command 'ssh root@90.90.90.201 "ip addr show  | grep 90.90.90.201 | grep -oE '[^ ]+$'"' finished with 0 after 0.329459190369s
Running 'ip -f inet -o addr show eth1 | awk '{print $4}' | cut -d / -f1'
[stdout] 90.90.90.205
Command 'ip -f inet -o addr show eth1 | awk '{print $4}' | cut -d / -f1' finished with 0 after 0.00443816184998s
Running 'echo 0 > /proc/sys/net/ipv4/icmp_echo_ignore_broadcasts'
Command 'echo 0 > /proc/sys/net/ipv4/icmp_echo_ignore_broadcasts' finished with 0 after 0.00205087661743s
Running 'ip link set eth1 allmulticast on'
Command 'ip link set eth1 allmulticast on' finished with 0 after 0.00298380851746s
Running 'ssh root@90.90.90.201 "ip route add 224.0.0.0/4 dev enp128s0f4d2"'
Command 'ssh root@90.90.90.201 "ip route add 224.0.0.0/4 dev enp128s0f4d2"' finished with 0 after 0.324348926544s
Running 'timeout 600 ssh root@90.90.90.201 "ping -I enp128s0f4d2 224.0.0.1 -c 1 | grep 90.90.90.205"'
[stdout] 64 bytes from 90.90.90.205: icmp_seq=1 ttl=64 time=0.148 ms
Command 'timeout 600 ssh root@90.90.90.201 "ping -I enp128s0f4d2 224.0.0.1 -c 1 | grep 90.90.90.205"' finished with 0 after 0.339205980301s
Running 'ssh root@90.90.90.201 "ip route del 224.0.0.0/4"'
Command 'ssh root@90.90.90.201 "ip route del 224.0.0.0/4"' finished with 0 after 0.333411931992s
Running 'echo 1 > /proc/sys/net/ipv4/icmp_echo_ignore_broadcasts'
Command 'echo 1 > /proc/sys/net/ipv4/icmp_echo_ignore_broadcasts' finished with 0 after 0.00197196006775s
Running 'ip link set eth1 allmulticast off'
Command 'ip link set eth1 allmulticast off' finished with 0 after 0.00307297706604s
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 1-multicast.py:ReceiveMulticastTest.test_multicast;Parameters-21f4

Not logging /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor (file does not exist)
Not logging /proc/pci (file does not exist)
Test results available in /root/avocado/job-results/job-2017-11-28T01.45-2b436b5

File: io/net/multicast.py [hunk 1]
```diff
@@ -79,6 +79,11 @@ def test_multicast(self):
         if process.system(cmd, shell=True, verbose=True,
                           ignore_status=True) != 0:
             self.fail("unable to set all mulicast option to test interface")
+        peer_route_add = "ip route add 224.0.0.0/4 dev %s" % self.peerif
```
Comment (narasimhan-v) [inline io/net/multicast.py]: I would rather reuse the variable `msg` instead of introducing another temporary variable. Except that, looks fine to me.

Comment (hathyaga) [reply io/net/multicast.py]: Why not keep that variable which tells the action being performed?

Comment (narasimhan-v) [reply io/net/multicast.py]: It is aboutuniformity

**Comment by hathyaga** [comment]
Version2 is sent upstream.
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/775

Closing this PR

**Pull request closed without merging**